sunkit_image/radial.py

```diff
@@ -577,3 +583,107 @@ def fnrgf(
+
+    Parameters
+    ----------
+    smap : `sunpy.map.Map`
+        The sunpy map to enhance.
+    radial_bin_edges : `astropy.units.Quantity`
+        A two-dimensional array of bin edges of size ``[2, nbins]`` where ``nbins`` is
+        the number of bins.
+    application_radius : `astropy.units.Quantity`, optional
+        The RHEF is applied to emission at radii above the application_radius.
+        Defaults to 0 solar radii.
+    upsilon : None or int or tuple of size 2, optional
+        A double-sided gamma function applied to the equalized histograms.
+        See Equation (4.15) in the thesis
+        Defaults to 0.35
+    method: {"inplace", "numpy", "scipy"}
+        A string describing which method to use for sorting.
+        Defaults to "numpy".
+
+    Returns
+    -------
+    `sunpy.map.Map`
+        A SunPy map that has had the RHEF applied to it.
+
+    References
+    ----------
+    * Gilly & Cranmer 2024, in prep.
+
+    * The implementation is highly inspired by this doctoral thesis:
+      Gilly, G. Spectroscopic Analysis and Image Processing of the Optically-Thin Solar Corona
+      https://www.proquest.com/docview/2759080511
+    """
+
+    # Get the radii for every pixel
+    map_r = find_pixel_radii(smap).to(u.R_sun)
+
+    if radial_bin_edges is None:
+        radial_bin_edges = equally_spaced_bins(0, 2, smap.data.shape[0] // 2)
+        radial_bin_edges *= u.R_sun
+
+    # Make sure bins are in the map.
+    if radial_bin_edges[1, -1] > np.max(map_r):
+        radial_bin_edges = equally_spaced_bins(
+            inner_value=radial_bin_edges[0, 0],
+            outer_value=np.max(map_r),
+            nbins=radial_bin_edges.shape[1],
+        )
+
+    # Select the sort method
+    if method == "inplace":
+        the_func = percentile_ranks_numpy_inplace
+    elif method == "numpy":
+        the_func = percentile_ranks_numpy
+    elif method == "scipy":
+        the_func = percentile_ranks_scipy
+    else:
+        msg = f"{method} is invalid. Allowed values are 'inplace', 'numpy', 'scipy'"
+        raise NotImplementedError(msg)
+
+    # Allocate storage for the filtered data
+    data = np.ones_like(smap.data)
+    meta = smap.meta
+
+    # Calculate the filter values for each radial bin.
+    for i in tqdm(range(radial_bin_edges.shape[1])):
+        # Identify the appropriate radial slice
+        here = np.logical_and(map_r >= radial_bin_edges[0, i], map_r < radial_bin_edges[1, i])
+        if application_radius is not None and application_radius > 0:
+            here = np.logical_and(here, map_r >= application_radius)
+
+        # Perform the filtering operation
+        data[here] = the_func(smap.data[here])
+        if upsilon is not None:
+            data[here] = apply_upsilon(data[here], upsilon)
+    new_map = sunpy.map.Map(data, meta)
+
+    if vignette not in [False, None]:
+        if vignette is True:
```

Comment:
Do we need both if statements?

Reply:
I think that this lets you either use the default value if you pass a True, or pass a quantity to control the value manually. Maybe there's a better way to do it?

Reply:
You can use https://docs.astropy.org/en/stable/api/astropy.units.quantity_input.html#astropy.units.quantity_input to ensforce the unit and we default to None as the default. Only apply it if users use the radius instead of True.